File: frontend/src/services/customerService.ts
// src/services/customerService.ts

import axiosInstance from "./axiosInstance";
import { Customer } from "../interfaces/Customer";

export const getAllCustomers = () => axiosInstance.get<Customer[]>("/customers");
export const getCustomerById = (id: number) =>
  axiosInstance.get<Customer>(`/customers/${id}`);
export const createCustomer = (customer: Customer) =>
  axiosInstance.post<Customer>("/customers", customer);
export const updateCustomer = (id: number, customer: Customer) =>
  axiosInstance.put<Customer>(`/customers/${id}`, customer);
export const deleteCustomer = (id: number) => axiosInstance.delete(`/customers/${id}`);

File: backend/src/main/java/com/example/backend/service/impliments/CustomerService.java
package com.example.backend.service.impliments;

import com.example.backend.persistence.dao.CustomerRepository;
import com.example.backend.persistence.entities.Customer;
import com.example.backend.service.interfaces.ICustomer;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class CustomerService implements ICustomer {

    private final CustomerRepository customerRepository;

    @Override
    public List<Customer> getAllCustomers() {
        return customerRepository.findAll();
    }

    @Override
    public Customer saveCustomer(Customer customer) {
        return customerRepository.save(customer);
    }

    @Override
    public Optional<Customer> getCustomerById(Long id) {
        return customerRepository.findById(id);
    }

    @Override
    public void deleteCustomer(Long id) {
        customerRepository.deleteById(id);
    }
}
